Bug: Running multiple conversations within one workflow issue

<img width="555" alt="Image" src="https://github.com/user-attachments/assets/ad2cbfc1-71e9-46e4-b5e5-36c410d517a0" />

**Describe the bug**

If we run multiple conversations (one after another) within the same workflow, only the first conversation is displayed in mesop.

**How to reproduce**
Include source code:

```python
import os
from typing import Any

from autogen.agentchat import ConversableAgent
from fastagency import UI
from fastagency.runtimes.ag2 import Workflow


llm_config = {
    "config_list": [
        {
            "model": "gpt-4o-mini",
            "api_key": os.getenv("OPENAI_API_KEY"),
        }
    ],
    "temperature": 0.8,
}

wf = Workflow()

def start_conversation(ui: UI) -> str:
    print("Starting repeatable process... PRINT")
    ui.text_message(
        sender="system",
        body="Starting conversation",
    )
    agent = ConversableAgent(
        name="Repeatable_Agent",
        system_message="You are a repeatable agent.",
        llm_config=llm_config,
    )

    response = agent.run(
        message="What is the capital of France?",
        summary_method="reflection_with_llm",
        max_turns=3,
    )

    response.process(ui)

    summary = response.summary
    print("Summary:", summary)

    return summary


@wf.register(name="simple_learning", description="Student and teacher learning chat")  # type: ignore[misc]
def simple_workflow(ui: UI, params: dict[str, Any]) -> str:    
    # In the first run, everything works fine
    summary = start_conversation(ui)
    summary += "\n\nRepeat...\n\n"
    # In the second run, messages are not displayed in mesop
    summary += start_conversation(ui)

    return summary
```

```console
2025-04-14 14:22:02,990 [INFO] Initialized MesopUI: <fastagency.ui.mesop.mesop.MesopUI object at 0x124547c90>
Starting repeatable process... PRINT
2025-04-14 14:22:03,098 [WARNING] Failed to render message: 'RunCompletionEvent' object has no attribute 'sender'
2025-04-14 14:22:03,098 [ERROR] home_page(): Error rendering home page: 'RunCompletionEvent' object has no attribute 'sender'
2025-04-14 14:22:03,100 [WARNING] Failed to render message: 'RunCompletionEvent' object has no attribute 'sender'
2025-04-14 14:22:03,100 [ERROR] home_page(): Error rendering home page: 'RunCompletionEvent' object has no attribute 'sender'
Summary: The capital of France is Paris. The user did not provide further input after the initial question.
Starting repeatable process... PRINT
2025-04-14 14:22:04,123 [WARNING] Failed to render message: 'RunCompletionEvent' object has no attribute 'sender'
2025-04-14 14:22:04,124 [ERROR] home_page(): Error rendering home page: 'RunCompletionEvent' object has no attribute 'sender'
```

--- a/fastagency/ui/console/console.py
+++ b/fastagency/ui/console/console.py
@@ -24,6 +24,7 @@ if TYPE_CHECKING:
     from autogen.events.agent_events import (
         ExecuteFunctionEvent,
         InputRequestEvent,
+        RunCompletionEvent,
         TerminationEvent,
         TextEvent,
         UsingAutoReplyEvent,
@@ -144,6 +145,10 @@ class ConsoleUI(MessageProcessorMixin, CreateWorkflowUIMixin):  # implements UI
 
     def visit_using_auto_reply(self, message: "UsingAutoReplyEvent") -> None:
         # Do nothing if it is of type UsingAutoReplyEvent
+        pass
+
+    def visit_run_completion(self, message: "RunCompletionEvent") -> None:
+        # We can ignore the RunCompletionEvent as we handle RunResponse already
         pass
 
     def visit_execute_function(self, message: "ExecuteFunctionEvent") -> None:

--- a/fastagency/ui/mesop/mesop.py
+++ b/fastagency/ui/mesop/mesop.py
@@ -39,6 +39,7 @@ if TYPE_CHECKING:
     from autogen.events.agent_events import (
         ExecuteFunctionEvent,
         InputRequestEvent,
+        RunCompletionEvent,
         TerminationEvent,
         TextEvent,
         UsingAutoReplyEvent,
@@ -242,6 +243,10 @@ class MesopUI(MessageProcessorMixin, CreateWorkflowUIMixin):  # UIBase
         self._publish(mesop_msg)
 
     def visit_using_auto_reply(self, message: "UsingAutoReplyEvent") -> None:
+        pass
+
+    def visit_run_completion(self, message: "RunCompletionEvent") -> None:
+        # We can ignore the RunCompletionEvent as we handle RunResponse already
         pass
 
     def visit_execute_function(self, message: "ExecuteFunctionEvent") -> None:

--- a/fastagency/ui/mesop/message.py
+++ b/fastagency/ui/mesop/message.py
@@ -1,12 +1,11 @@
 import json
 import random
 from collections.abc import Iterable, Iterator
-from typing import Callable, Optional
+from typing import TYPE_CHECKING, Callable, Optional
 from uuid import UUID, uuid4
 
 import mesop as me
 import mesop.labs as mel
-from autogen.events.agent_events import TextEvent, UsingAutoReplyEvent
 
 from fastagency.helpers import jsonify_string
 
@@ -32,6 +31,13 @@ from .mesop import MesopMessage
 from .send_prompt import get_more_messages, send_user_feedback_to_autogen
 from .styles import MesopHomePageStyles, MesopMessageStyles
 from .timer import wakeup_component
+
+if TYPE_CHECKING:
+    from autogen.events.agent_events import (
+        RunCompletionEvent,
+        TextEvent,
+        UsingAutoReplyEvent,
+    )
 
 logger = get_logger(__name__)
 
@@ -208,7 +214,7 @@ class MesopGUIMessageVisitor(MessageProcessorMixin):
             if inner_callback:
                 inner_callback()
 
-    def visit_text(self, message: TextEvent) -> None:
+    def visit_text(self, message: "TextEvent") -> None:
         content = message.content.content
         self.visit_default(
             message,
@@ -216,13 +222,17 @@ class MesopGUIMessageVisitor(MessageProcessorMixin):
             style=self._styles.message.text,
         )
 
-    def visit_using_auto_repy(self, message: UsingAutoReplyEvent) -> None:
+    def visit_using_auto_repy(self, message: "UsingAutoReplyEvent") -> None:
         content = None
         self.visit_default(
             message,
             content=content,
             style=self._styles.message.system,
         )
+
+    def visit_run_completion(self, message: "RunCompletionEvent") -> None:
+        # We can ignore the RunCompletionEvent as we handle RunResponse already
+        pass
 
     def visit_text_message(self, message: TextMessage) -> None:
         content = message.body if message.body else ""

